### PR TITLE
DD_SERVICE_MAPPING in extension

### PR DESCRIPTION
**Pull request by zARODz11z**
### What does this PR do?
allows users to remap the service name on inferred spans created in dd lambda js. Pairs with https://github.com/DataDog/documentation/pull/18144

### Motivation
SLS-3501

### Describe how to test/QA your changes
unit

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/serverless-DD-SERVICE-MAPPING-594cc2cb7d090473.yaml
Comment (urseberry) [inline releasenotes/notes/serverless-DD-SERVICE-MAPPING-594cc2cb7d090473.yaml:4]:
```suggestion
    `DD_SERVICE_MAPPING` can be used to rename Serverless inferred spans' service names.
```

**Review by urseberry — APPROVED** [review]
The line in the release note is a bit confusing to me. If I'm interpreting it correctly, the service names belong to the inferred spans.

Approving so I don't hold up the release process.

**Review by ogaca-dd — APPROVED** [review]

**Comment by zARODz11z** [comment]
sls integration test failure is known issue

File: pkg/config/config.go
Comment (sgnn7) [inline pkg/config/config.go:1169]: Any reason why we wouldn't have this scoped to `serverless.` prefix?

**Review by sgnn7 — APPROVED** [review]
LGTM for `pkg/config/config.go` - just a small question/note

File: pkg/config/config.go
Comment (zARODz11z) [reply pkg/config/config.go:1169]: honestly, no
I just used the above lines as a guide to get this working with the cfg pkg and just left it there. Prob better to add the prefix for now. Ty

File: pkg/serverless/trace/inferredspan/span_enrichment.go
Comment (duncanista) [inline pkg/serverless/trace/inferredspan/span_enrichment.go:51]: Can we move this to the tests instead of having it here?

**Review by duncanista — CHANGES_REQUESTED** [review]
LGTM – nitpick around functions that are used for tests being inside `span_enrichment.go`

File: pkg/serverless/trace/inferredspan/span_enrichment_test.go
Comment (joeyzhao2018) [inline pkg/serverless/trace/inferredspan/span_enrichment_test.go:27]: can we add empty string case and ",a:b," case?

File: pkg/serverless/trace/inferredspan/span_enrichment.go
Comment (zARODz11z) [reply pkg/serverless/trace/inferredspan/span_enrichment.go:51]: I know I moved them in the python implementation but I think its best to have them here, it provides a way to change and access the state of serviceMapping within the tests. Otherwise if I moved them I would have to export serviceMapping (i.e. make it capitalized) and that could lead to other issues down the road of that variable being altered in other ways we didn't intend to. LMKWYK

Comment (zARODz11z) [reply pkg/serverless/trace/inferredspan/span_enrichment.go:51]: hmm well I guess someone could use SetServiceMapping and GetServiceMapping anyways to change serviceMapping since they are exported functions but at least this way we know they will interact with it through these defined ways

File: pkg/serverless/trace/inferredspan/span_enrichment_test.go
Comment (zARODz11z) [reply pkg/serverless/trace/inferredspan/span_enrichment_test.go:27]: added cases

**Review by duncanista — APPROVED** [review]

**Pull request merged**
